routes/eth.go

```diff
@@ -241,41 +260,124 @@ func (fes *APIServer) SubmitETHTx(ww http.ResponseWriter, req *http.Request) {
 		}
 	}
 
-	// Ensure the transaction mined
-	if ethTx == nil || ethTx.BlockHeight < 0 {
-		_AddBadRequestError(ww, "SubmitETHTx: Transaction failed to mine")
+	bitcloutTxnHash, err := fes.finishETHTx(ethTxLog)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("SubmitETHTx: Failed: %v", err))
 		return
 	}
 
+	res := SubmitETHTxResponse{
+		BitCloutTxnHash: bitcloutTxnHash.String(),
+	}
+	if err := json.NewEncoder(ww).Encode(res); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("SubmitETHTx: Problem encoding response: %v", err))
+		return
+	}
+}
+
+// 1. Validate the transaction mined
+// 2. Calculate the nanos to send
+// 3. Send the nanos
+// 4. Record the successful send
+func (fes *APIServer) finishETHTx(ethTxLog *ETHTxLog) (txnHash *lib.BlockHash, _err error) {
+	// Ensure the transaction mined
+	if ethTxLog.Tx.BlockHeight < 0 {
+		return nil, errors.New("Transaction failed to mine")
+	}
+
 	// Fetch buy bitclout basis points fee
 	feeBasisPoints, err := fes.GetBuyBitCloutFeeBasisPointsResponseFromGlobalState()
 	if err != nil {
-		_AddBadRequestError(ww, fmt.Sprintf("SubmitETHTx: Error getting buy fee basis points: %v", err))
-		return
+		return nil, errors.New(fmt.Sprintf("Error getting buy fee basis points: %v", err))
 	}
 
 	// Calculate nanos purchased
-	totalWei := big.NewFloat(0).SetInt(ethTx.Total)
+	totalWei := big.NewFloat(0).SetInt(ethTxLog.Tx.Total)
 	totalEth := big.NewFloat(0).Quo(totalWei, big.NewFloat(1e18))
 	nanosPurchased := fes.GetNanosFromETH(totalEth, feeBasisPoints)
 
-	bitcloutTxnHash, err := fes.SendSeedBitClout(pkBytes, nanosPurchased, true)
+	bitcloutTxnHash, err := fes.SendSeedBitClout(ethTxLog.PublicKey, nanosPurchased, true)
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("Error sending BitClout: %v", err))
+	}
+
+	// Record successful transaction in global state
+	ethTxLog.BitCloutTxnHash = bitcloutTxnHash.String()
+	globalStateKey := GlobalStateKeyETHPurchases(ethTxLog.Tx.Hash)
+	globalStateVal := bytes.NewBuffer([]byte{})
+	err = gob.NewEncoder(globalStateVal).Encode(ethTxLog)
```

Comment:
In the Wyre flow, we mark the transaction as paid out before we send money and then unmark it if there is an error sending the money. If I recall correctly, the rationale was that we won't want someone to be able to continually get money if there's an issue with global state for some reason. I think it's okay here, but just wanted to mention it here.